yjit/src/codegen.rs

```diff
@@ -7116,10 +7116,11 @@ fn gen_send_iseq(
     let iseq_has_block_param = unsafe { get_iseq_flags_has_block(iseq) };
     let arg_setup_block = captured_opnd.is_some(); // arg_setup_type: arg_setup_block (invokeblock)
     let kw_splat = flags & VM_CALL_KW_SPLAT != 0;
-    let splat_call = flags & VM_CALL_ARGS_SPLAT != 0;
 
     let forwarding_call = unsafe { rb_get_iseq_flags_forwardable(iseq) };
 
+    let splat_call = (flags & VM_CALL_ARGS_SPLAT != 0) && !forwarding_call;
```

Comment:
This is because this call site (this send instruction) is a forwarding call?

Pls add more comments for us laypeople